examples/hello-world-flask/Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3.10-alpine
+
+WORKDIR /app
+
+RUN apk add --no-cache gcc musl-dev python3-dev libffi-dev openssl-dev cargo g++
+
+ENV PYTHONFAULTHANDLER=1 \
+  PYTHONUNBUFFERED=1 \
+  PYTHONHASHSEED=random \
+  PIP_NO_CACHE_DIR=off \
+  PIP_DISABLE_PIP_VERSION_CHECK=on \
+  PIP_DEFAULT_TIMEOUT=100 \
+  POETRY_VERSION=1.3.2
+
+# This project is poetic.
+RUN pip install "poetry==$POETRY_VERSION"
+# This image is single purpose, so we won't need to compartmentalize Py deps in virtualenvs.
+RUN poetry config virtualenvs.create false
+
+# Copy the distro into the image and install deps.
+COPY README.md pyproject.toml poetry.lock ./
+COPY src/ ./src/
+RUN poetry install
+
+# Copy the examples into the image and install deps.
+COPY examples/ ./examples/
+RUN cd ./examples/hello-world-flask && poetry install
+
+# From this point forward, we're operating on the flask example.
+WORKDIR /app/examples/hello-world-flask
+CMD ["poetry", "run", "opentelemetry-instrument", "flask", "run", "--host", "0.0.0.0"]
```

Comment:
Apparently we need to pass the host in the command line argument even though it's specified in the app... many thanks to [ this StackOverflow ](https://stackoverflow.com/questions/30323224/deploying-a-minimal-flask-app-in-docker-server-connection-issues/58138250)for guidance